frontends/verilog/verilog_parser.y

```diff
@@ -1541,18 +1541,60 @@ list_of_specparam_assignments:
 specparam_assignment:
 	ignspec_id '=' ignspec_expr ;
 
-ignspec_opt_cond:
-	TOK_IF '(' ignspec_expr ')' | %empty;
-
 path_declaration :
 	simple_path_declaration ';'
 	// | edge_sensitive_path_declaration
-	// | state_dependent_path_declaration
+	| state_dependent_path_declaration ';'
 	;
 
 simple_path_declaration :
-	ignspec_opt_cond parallel_path_description '=' path_delay_value |
-	ignspec_opt_cond full_path_description '=' path_delay_value
+	parallel_path_description '=' path_delay_value |
+	full_path_description '=' path_delay_value
+	;
+
+state_dependent_path_declaration:
+	TOK_IF '(' module_path_expression ')' simple_path_declaration
+	// | TOK_IF '(' module_path_expression ')' edge_sensitive_path_declaration
+	| TOK_IFNONE simple_path_declaration
+	;
+
+module_path_expression:
+	module_path_primary
+	// Flatten out unary_operator to avoid shift/reduce conflict
+	| '!' attr module_path_primary { delete $2; }
```

Comment:
Based on the existing code, these need to be `freeattr($2);` instead to avoid a memory leak (not that I'm sure why they're being deleted).

Reply:
All these rules are only used in `ignored_specify_item` which only applies when `-specify` is off so we discard attributes everything including these attributes that auto-allocate